src/AuthProvider.jsx

```diff
@@ -8,29 +8,27 @@ export const useAuth = () => {
 
 export const AuthProvider = ({ children}) => {
     const [isLoggedIn, setIsLoggedIn] = useState(false);
-    const [userData, setUserData] = useState({ firstName: "", lastName: "", role: "" })
-    const isToken = sessionStorage.getItem('isToken');
-
+    const [userData, setUserData] = useState(JSON.parse(sessionStorage.getItem('user')) || {});
+    
     useEffect(() => {
-        if (isToken) {
+        if (userData.token) {
             setIsLoggedIn(true);
         } else {
             setIsLoggedIn(false);
         }
-    }, [isToken]);
+    }, [userData.token]);
 
-    const loginAction = (status, data) => {
-        sessionStorage.setItem('isToken', status);
-        setIsLoggedIn(true);
-        setUserData({ firstName: data.firstName, lastName: data.lastName, role: data.role });
+    const loginUser = (status, data) => {
```

Comment:
These two functions will benefit from names that better describe their actions. `setUserData` or `removeUserData` are good options. `setUserDataToSession` could also be a good candidate but your function does more than just set session data. It also sets the user data to state.